### PR TITLE
fix(react-router): ensure links and scripts are available on routes created with `createRoute`

### PR DESCRIPTION
**Pull request by GregoryCollett**
We found that creating a route with `createRoute` would not expose links and scripts where as when creating a route with `createFileRoute` the link and script properties would exist as expected. We expect the behaviour to be the same across the board.

### CODE REVIEW

**Comment by schiller-manuel** [comment]
a e2e test in one of the e2e start tests would be nice.

File: packages/react-router/src/router.ts [hunk 1]
```diff
@@ -2528,6 +2528,8 @@ export class Router<
                             loaderData,
                           })
                           const meta = headFnContent?.meta
+                          const links = headFnContent?.links
```
Comment (schiller-manuel) [inline packages/react-router/src/router.ts:2531]: I don't understand why this worked in `createFileRoute` but in in `createRoute`. can you explain this please?

**Review by SeanCassiere — APPROVED** [review]

File: packages/react-router/src/router.ts (hunk 1 above)
Comment (GregoryCollett) [reply packages/react-router/src/router.ts:2531]: Sorry, I haven't had time to fully evaluate this but agree it didn't seem immediately apparent why this worked using createRootRoute and createFileRoute vs createRoute. I have to assume they go through the same path and it seems file route calls createRoute under the hood... I will explore this more but as I say I haven't yet found the time to give this an in depth diagnosis beyond the initial findings.

**Comment by schiller-manuel** [comment]
thanks for checking this

**Pull request merged**